protos/flyteidl/event/event.proto

```diff
@@ -151,13 +151,54 @@ message TaskExecutionEvent {
     // The version field should be incremented when metadata changes across the duration of an individual phase.
     uint32 phase_version                  = 12;    
 
+    // An optional explanation for the phase transition.
+    string reason                         = 13;
+
+
+    // A predefined yet extensible Task type identifier. If the task definition is already registered in flyte admin
+    // this type will be identical, but not all task executions necessarily use pre-registered definitions and this
+    // type is useful to render the task in the UI, filter task executions, etc.
+    string task_type                      = 14;
+
     // Metadata around how a task was executed.
     TaskExecutionMetadata metadata        = 16;
 }
 
+// This message contains metadata about external resources produced or used by a specific task execution.
+message ExternalResourceInfo {
+
+    // Identifiers for external resources created by this task execution, for example Qubole query ID or presto query ids.
+    repeated string external_ids = 1;
```

Comment:
If we have a repeated field [here](https://github.com/flyteorg/flyteidl/pull/119/files#diff-61c99ca77e3f94bf6d64036e11e8851dd5a9defc8fb5f1d385af4bfc52b9e9f4R197) we don't need this to be repeated, right? this can include other things if the plugin wants to (maybe cost related, maybe remote cluster/region info ... etc.) that we can open up later

Reply:
done